docs/io/ascii/ecsv.rst

```diff
@@ -0,0 +1,471 @@
+  # - {name: a, datatype: int64}
+  # - {name: b, datatype: float64}
+  # - {name: c, datatype: string}
+  # - {name: c.mask, datatype: bool}
+  # meta: !!omap
+  # - __serialized_columns__:
+  #     c:
+  #       __class__: astropy.table.column.MaskedColumn
+  #       data: !astropy.table.SerializedColumn {name: c}
+  #       mask: !astropy.table.SerializedColumn {name: c.mask}
+  # schema: astropy-2.0
+  a b c c.mask
+  "" 1.0 "" False
+  2 2.0 d True
+  3 "" e False
+
+When you read this back in, both the empty (zero-length) string and the masked
+``'d'`` value in the column ``'c'`` will be preserved.
+
+..
+  EXAMPLE END
+
+.. _ecsv_format_mixin_columns:
+
+Mixin Columns
+-------------
+
+It is possible to store not only standard `~astropy.table.Column` and
+`~astropy.table.MaskedColumn` objects to ECSV but also the following
+:ref:`mixin_columns`:
+
+- `astropy.time.Time`
+- `astropy.time.TimeDelta`
+- `astropy.units.Quantity`
+- `astropy.coordinates.Latitude`
+- `astropy.coordinates.Longitude`
+- `astropy.coordinates.Angle`
+- `astropy.coordinates.Distance`
+- `astropy.coordinates.EarthLocation`
+- `astropy.coordinates.SkyCoord`
+- `astropy.table.NdarrayMixin`
+- Coordinate representation types such as `astropy.coordinates.SphericalRepresentation`
+
+In general, a mixin column may contain multiple data components as well as
+object attributes beyond the standard `~astropy.table.Column` attributes like
+``format`` or ``description``. Storing such mixin columns is done by replacing
+the mixin column with column(s) representing the underlying data component(s)
+and then inserting metadata which informs the reader of how to reconstruct the
+original column. For example, a `~astropy.coordinates.SkyCoord` mixin column in
+``'spherical'`` representation would have data attributes ``ra``, ``dec``,
+``distance``, along with object attributes like ``representation_type`` or
+``frame``.
+
+..
+  EXAMPLE START
+  Writing a Table with a SkyCoord Column in ECSV Format
+
+This example demonstrates writing a `~astropy.table.QTable` that has `~astropy.time.Time`
+and `~astropy.coordinates.SkyCoord` mixin columns::
+
+  >>> from astropy.coordinates import SkyCoord
+  >>> import astropy.units as u
+  >>> from astropy.table import QTable
+
+  >>> sc = SkyCoord(ra=[1, 2] * u.deg, dec=[3, 4] * u.deg)
+  >>> sc.info.description = 'flying circus'
+  >>> q = [1, 2] * u.m
+  >>> q.info.format = '.2f'
+  >>> t = QTable()
+  >>> t['c'] = [1, 2]
+  >>> t['q'] = q
+  >>> t['sc'] = sc
+
+  >>> t.write('my_data.ecsv')  # doctest: +SKIP
+
+The contents of ``my_data.ecsv`` are below::
+
+  # %ECSV 1.0
+  # ---
+  # datatype:
+  # - {name: c, datatype: int64}
+  # - {name: q, unit: m, datatype: float64, format: .2f}
+  # - {name: sc.ra, unit: deg, datatype: float64}
+  # - {name: sc.dec, unit: deg, datatype: float64}
+  # meta: !!omap
+  # - __serialized_columns__:
+  #     q:
+  #       __class__: astropy.units.quantity.Quantity
+  #       __info__: {format: .2f}
```

Comment:
At some point it would be nice to get rid of this duplication of `format` here and in the general description - but for another PR!